ntp-daemon/src/main.rs

```diff
@@ -1,169 +1,165 @@
-
-                        let adjust_type = controller.update(
-                            clock_select.system_offset,
-                            clock_select.system_jitter,
-                            clock_select.system_peer_snapshot.root_delay,
-                            clock_select.system_peer_snapshot.root_dispersion,
-                            clock_select.system_peer_snapshot.leap_indicator,
-                            clock_select.system_peer_snapshot.time,
-                        );
-
-                        // Handle situations needing extra processing
-                        match adjust_type {
-                            ClockUpdateResult::Panic => {
-                                panic!("Unusually large clock step suggested, please manually verify system clock and reference clock state and restart if appropriate.")
-                            }
-                            ClockUpdateResult::Step => {
-                                waiting_for_reset.append(&mut peers);
-                                last_reset_index += 1;
-                                reset_tx.send_replace(last_reset_index);
-                            }
-                            _ => {}
-                        }
-
-                        // Handle updating system snapshot
-                        match adjust_type {
-                            ClockUpdateResult::Ignore => {},
-                            _ => {
-                                let mut system_snapshot = *system_rx.borrow();
-                                system_snapshot.poll_interval = controller.preferred_poll_interval();
-                                system_snapshot.leap_indicator = clock_select.system_peer_snapshot.leap_indicator;
-                                system_tx.send(system_snapshot).expect("System snapshot mechanism failed");
-                            }
-                        }
-                    }
-                    None => info!("filter and combine did not produce a result"),
+        // remove snapshots from previous iteration
+        snapshots.clear();
+
+        // add all valid measurements to our list of snapshots
+        snapshots.extend(peers.iter().filter_map(PeerState::get_snapshot));
+
+        let ntp_instant = NtpInstant::now();
+        let system_poll = global_system_snapshot.read().await.poll_interval;
+        let result = FilterAndCombine::run(&config, &snapshots, ntp_instant, system_poll);
+
+        let clock_select = match result {
+            Some(clock_select) => clock_select,
+            None => {
+                info!("filter and combine did not produce a result");
+                continue;
+            }
+        };
+
+        let offset_ms = clock_select.system_offset.to_seconds() * 1000.0;
+        let jitter_ms = clock_select.system_jitter.to_seconds() * 1000.0;
+        info!(offset_ms, jitter_ms, "system offset and jitter");
+
+        let adjust_type = controller.update(
```

Comment:
Why not just pass `&clock_select` to `update()`? This looks silly.

Reply:
I believe the idea here was to not pass anything into the clock that it doesn't need, to make the data flow easier (and e.g. to spot that certain fields are never read). @davidv1992 do you have an opinion here. It does look silly, but it's also harmless so I don't have strong feelings either way

Reply:
I think this is a remnant from an earlier stage in development as this logic was developed partially in parallel with the clock_select struct, and have no problem just passing the clock select input directly, as it indeed looks somewhat silly

Reply:
so, i tried this and the alternative is worse; we still need to unpack all those fields. We also have a bunch of tests that supply all these arguments. This does not really spark joy
```rust
let &FilterAndCombine {
    system_offset: offset,
    system_jitter: jitter,
    system_peer_snapshot: snapshot,
} = filter_and_combine;

let PeerSnapshot {
    time: last_peer_update,
    leap_indicator: leap_status,
    root_delay,
    root_dispersion,
    ..
} = snapshot;
```

Reply:
Ok, in that case I suggest we just keep it as is.